sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNodeSuite.scala

```diff
@@ -40,10 +41,38 @@ class SqlScriptingExecutionNodeSuite extends SparkFunSuite with SharedSparkSessi
       Origin(startIndex = Some(0), stopIndex = Some(description.length)),
       isInternal = false)
 
+  case class TestWhileCondition(description: String)
+    extends SingleStatementExec(
+      parsedPlan = Project(Seq(Alias(Literal(true), description)()), OneRowRelation()),
+      Origin(startIndex = Some(0), stopIndex = Some(description.length)),
+      isInternal = false)
+
+  case class TestWhile(
+      condition: SingleStatementExec,
+      body: CompoundBodyExec,
+      reps: Int)
+    extends WhileStatementExec(condition, body, spark) {
+
+    private var callCount: Int = 0
+
+    override def evaluateBooleanCondition(
+      session: SparkSession,
```

Comment:
nit: the indentation should be the other way around - parameters are indented 4 spaces compared to the function, whereas the function code is indented 2 spaces.